### DISCUSSION TITLE
config files for Bulex heater (and/or MIgo thermostat) somewhere reverse engineered already?

### DISCUSSION THREAD

**Discussion by netweaver1970** [john30/ebusd#527] [Q&A]
Hi,
Last year I installed the EBUS Adapter 3 + Wemos D1 in a more manual install of HomeAssistant. I got something working by then, but the difficulty of finding all correct values + other issues made me to freeze the project.
Today I got the new HA integration working (-sometimes-, see new issue for this) in a proper Add-on fashion on a new HASS-OS install, kudos for @john30, so I got renewed energy to finally try to tame this beast and get all out what I would like + control it as I want.

But I still have the same issues as last year, my Bulex Themacondens heater is not properly recognized so the config falls back to a (generic) Vaillant BAI. And my thermostat is not recognized at all.
Did anyone work on custom config files for these (Belgian) Heaters? Below is the snippet from the log.

Address 3/8 is the Bulex ThemaCondens heater, Address 10/15 is the  Bulex Migo (aka Vaillant vSmart, aka Saunier Duval Migo, aka older version of the Netatmo Smart thermostat).

```
address 03: master #11
address 08: slave #11, scanned "MF=Vaillant;ID=BAI00;SW=0602;HW=3302"
address 10: master #2
address 15: slave #2, scanned "MF=Vaillant;ID=SCC00;SW=2200;HW=3903"
address 31: master #8, ebusd
address 36: slave #8, ebusd
```

thanks.

**Comment by JensVanhooydonck** [comment]
Having the same issue with 
 ```Vaillant;SCC00;0200;3903```

**Comment by DiaMontPhil** [comment]
Raspberry Pi 3B + v3 ebus adapter on GPIO/ttyAMA0
Heater : Bulex ThemaCondens with ExaMaster

ebusctl info:
version: ebusd 22.3.p20220508
...
device: /dev/ttyAMA0, enhanced
access: *
signal: acquired
symbol rate: 6
max symbol rate: 554
reconnects: 4
masters: 4
messages: 214
conditional: 3
poll: 0
update: 10
address 00: master #1
address 03: master #11
address 08: slave #11, scanned "MF=Vaillant;ID=BAI00;SW=0701;HW=3302", loaded "vaillant/bai.308523.inc", "vaillant/08.bai.csv"
address 10: master #2
address 31: master #8, ebusd
address 36: slave #8, ebusd

Address 3/8 is the Bulex ThemaCondens heater (same remark as @netweaver1970)
The ExaMaster is not recognized at all. This is expected as it is not in @John30 database. 

I'm new to eBus, even after reading many hours about it.

Communication is present: ebusctl listen:
listen started

bai DateTime = nosignal;-:-:-;-.-.-;18.625
bai DateTime = nosignal;-:-:-;-.-.-;18.438
bai Status01 = 26.5;26.5;18.438;-;-;off
bai Status01 = 26.5;26.5;18.625;-;-;off
broadcast outsidetemp = 16.438
broadcast vdatetime = 23:08:00;27.08.2022
bai DateTime = nosignal;-:-:-;-.-.-;18.625
broadcast outsidetemp = 16.625

Even if not all messages are decyphered, the broadcast outsidetemperature is nice to have. (The thermometer is connected by wire to the Bulex ThemaCondens).
I started to integrate all systems into OpenHab. The bridge is communicating, but no outsidetemp is available (nothing after the automatic scanner) or manually added.

Is there anyone else with similar problems/configurations ?

**Comment by DiaMontPhil** [comment]
After update of ebusd files:
ebusctl info:

version: ebusd 22.3.p20220508
update check: revision v22.3 available
device: /dev/ttyAMA0, enhanced
access: *
signal: acquired
symbol rate: 8
max symbol rate: 93
reconnects: 1
masters: 3
messages: 226
conditional: 3
poll: 0
update: 10
address 03: master #11
address 08: slave #11, scanned "MF=Vaillant;ID=BAI00;SW=0701;HW=3302", loaded "vaillant/bai.308523.inc", "vaillant/08.bai.csv"
address 10: master #2
address 15: slave #2, scanned "MF=Vaillant;ID=SBX00;SW=0326;HW=4802"
address 31: master #8, ebusd
address 36: slave #8, ebusd

Address 3/8 is still the Bulex ThemaCondens heater.
Address 10/15 is recognized. In reality it is the Examaster (Bulex).

I also flashed the ebus v3 pic with latest firmware.

I rebooted the system a few minutes ago.

in de log file there are a lot of the following messages:
2022-08-28 10:44:02.637 [bus error] device status: unexpected available enhanced byte 2

ebusctl scan result 
08;Vaillant;BAI00;0701;3302; ;21;13;3700100088;4816;100053;27
15;Vaillant;SBX00;0326;4802;21;17;16;0020128487;0082;005147;N2

ebusctl find -a
bai AccessoriesOne = no data stored
bai AccessoriesOne = no data stored
bai AccessoriesTwo = no data stored
bai AccessoriesTwo = no data stored
bai ACRoomthermostat = no data stored
bai APCComStatus = no data stored (message not available due to condition)
bai APCLegioProtection = no data stored (message not available due to condition)
bai APCLegioProtection = no data stored (message not available due to condition)
bai averageIgnitiontime = no data stored
bai BlockTimeHcMax = no data stored
bai BlockTimeHcMax = no data stored
bai BoilerType = no data stored
bai ChangesDSN = no data stored
bai CirPump = no data stored
bai clearerrorhistory = no data stored
bai CounterStartattempts1 = no data stored
bai CounterStartattempts2 = no data stored
bai CounterStartAttempts3 = no data stored
bai CounterStartAttempts4 = no data stored
bai currenterror = -;-;-;-;-
bai DateTime = nosignal;-:-:-;-.-.-;20.438
bai dcfState = no data stored
bai DCFTimeDate = no data stored
bai DCRoomthermostat = no data stored
bai DeactivationsIFC = no data stored
bai DeactivationsTemplimiter = no data stored
bai DeltaFlowReturnMax = no data stored
bai DisplayMode = no data stored
bai DSN = no data stored
bai DSNOffset = no data stored
bai DSNOffset = no data stored
bai DSNStart = no data stored
bai EBusHeatcontrol = no data stored
bai EbusSourceOn = no data stored
bai EbusVoltage = no data stored
bai errorhistory = no data stored
bai expertlevel_ReturnTemp = no data stored
bai ExternalFaultmessage = no data stored
bai externalFlowTempDesired = no data stored
bai externalHwcSwitch = no data stored
bai ExternGasvalve = no data stored
bai ExtFlowTempDesiredMin = no data stored
bai ExtStorageModulCon = no data stored
bai extWP = no data stored
bai FanHours = no data stored
bai FanMaxSpeedOperation = no data stored
bai FanMinSpeedOperation = no data stored
bai FanPWMSum = no data stored
bai FanPWMTest = no data stored
bai FanSpeed = no data stored
bai FanSpeedOffsetMax = no data stored
bai FanSpeedOffsetMax = no data stored
bai FanSpeedOffsetMin = no data stored
bai FanSpeedOffsetMin = no data stored
bai FanStarts = no data stored
bai Flame = no data stored
bai FlameSensingASIC = no data stored
bai FloorHeatingContact = no data stored
bai FlowsetHcMax = no data stored
bai FlowsetHcMax = no data stored
bai FlowsetHwcMax = no data stored
bai FlowsetHwcMax = no data stored
bai FlowSetPotmeter = no data stored
bai FlowTemp = no data stored
bai FlowTempDesired = no data stored
bai FlowTempMax = no data stored
bai Fluegasvalve = no data stored
bai FluegasvalveOpen = no data stored
bai Gasvalve3UC = no data stored
bai Gasvalve = no data stored
bai GasvalveASICFeedback = no data stored
bai GasvalveUC = no data stored
bai GasvalveUCFeedback = no data stored
bai HcHours = no data stored
bai HcPumpMode = no data stored
bai HcPumpMode = no data stored
bai HcPumpStarts = no data stored
bai HcStarts = no data stored
bai HcUnderHundredStarts = no data stored
bai HeatingSwitch = no data stored
bai HoursTillService = no data stored
bai HoursTillService = no data stored
bai HwcDemand = no data stored
bai HwcHours = no data stored
bai HwcImpellorSwitch = no data stored
bai HwcPostrunTime = no data stored
bai HwcPostrunTime = no data stored
bai HwcSetPotmeter = no data stored
bai HwcStarts = no data stored
bai HwcSwitch = no data stored
bai HwcTemp = no data stored
bai HwcTempDesired = no data stored
bai HwcTempMax = no data stored
bai HwcTempMax = no data stored
bai HwcTypes = no data stored
bai HwcUnderHundredStarts = no data stored
bai HwcWaterflow = no data stored
bai HwcWaterflowMax = no data stored
bai Ignitor = no data stored
bai InitialisationEEPROM = no data stored
bai IonisationVoltageLevel = no data stored
bai maintenancedata_HwcTempMax = no data stored
bai maxIgnitiontime = no data stored
bai minIgnitiontime = no data stored
bai ModulationTempDesired = no data stored
bai OutdoorstempSensor = no data stored
bai OverflowCounter = no data stored
bai ParamToken = no data stored
bai PartloadHcKW = no data stored
bai PartloadHcKW = no data stored
bai PartloadHwcKW = no data stored
bai PartloadHwcKW = no data stored
bai PartnumberBox = no data stored
bai PositionValveSet = no data stored
bai PowerValue = no data stored
bai PrAPSCounter = no data stored
bai PrAPSSum = no data stored
bai PrEnergyCountHc1 = no data stored
bai PrEnergyCountHc1 = no data stored
bai PrEnergyCountHc2 = no data stored
bai PrEnergyCountHc2 = no data stored
bai PrEnergyCountHc3 = no data stored
bai PrEnergyCountHc3 = no data stored
bai PrEnergyCountHwc1 = no data stored
bai PrEnergyCountHwc1 = no data stored
bai PrEnergyCountHwc2 = no data stored
bai PrEnergyCountHwc2 = no data stored
bai PrEnergyCountHwc3 = no data stored
bai PrEnergyCountHwc3 = no data stored
bai PrEnergySumHc1 = no data stored
bai PrEnergySumHc1 = no data stored
bai PrEnergySumHc2 = no data stored
bai PrEnergySumHc2 = no data stored
bai PrEnergySumHc3 = no data stored
bai PrEnergySumHc3 = no data stored
bai PrEnergySumHwc1 = no data stored
bai PrEnergySumHwc1 = no data stored
bai PrEnergySumHwc2 = no data stored
bai PrEnergySumHwc2 = no data stored
bai PrEnergySumHwc3 = no data stored
bai PrEnergySumHwc3 = no data stored
bai ProductionByte = no data stored
bai PrVortexFlowSensorValue = no data stored
bai PumpHours = no data stored
bai PumpHwcFlowNumber = no data stored
bai PumpHwcFlowSum = no data stored
bai RemainingBoilerblocktime = no data stored
bai ReturnRegulation = no data stored
bai ReturnRegulation = no data stored
bai ReturnTemp = no data stored
bai ReturnTempMax = no data stored
bai SecondPumpMode = no data stored
bai SecondPumpMode = no data stored
bai SerialNumber = no data stored
bai SetFactoryValues = no data stored
bai SetFactoryValues = no data stored
bai SetMode = auto;0.0;54.0;-;1;0;0;0;0;0
bai SHEMaxDeltaHwcFlow = no data stored
bai SHEMaxFlowTemp = no data stored
bai SolPostHeat = no data stored
bai SolPostHeat = no data stored
bai Statenumber = no data stored
bai Status01 = 25.5;26.0;20.250;-;-;off
bai Status02 = off;20;61.0;90;50.0
bai Status16 = no data stored
bai Status = no data stored
bai StatusCirPump = no data stored
bai Storageloadpump = no data stored
bai StorageLoadPumpHours = no data stored
bai StorageloadPumpStarts = no data stored
bai StorageLoadTimeMax = no data stored
bai StorageLoadTimeMax = no data stored
bai StoragereleaseClock = no data stored
bai StorageTemp = no data stored
bai StorageTempDesired = no data stored
bai StorageTempMax = no data stored
bai TargetFanSpeed = no data stored
bai TargetFanSpeedOutput = no data stored
bai TempDiffBlock = no data stored
bai TempDiffFailure = no data stored
bai TempGradientFailure = no data stored
bai Templimiter = no data stored
bai TemplimiterWithNTC = no data stored
bai TempMaxDiffExtTFT = no data stored
bai Testbyte = no data stored
bai TimerInputHc = no data stored
bai ValveMode = no data stored
bai ValveMode = no data stored
bai ValveStarts = no data stored
bai VolatileLockout = no data stored
bai VolatileLockoutIFCGV = no data stored
bai VortexFlowSensor = no data stored
bai WarmstartDemand = no data stored
bai WarmstartOffset = no data stored
bai WarmstartOffset = no data stored
bai WaterHcFlowMax = no data stored
bai WaterPressure = no data stored
bai WaterpressureBranchControlOff = no data stored
bai WaterpressureMeasureCounter = no data stored
bai WaterpressureVariantSum = no data stored
bai WP = no data stored
bai WPPostrunTime = no data stored
bai WPPostrunTime = no data stored
bai WPSecondStage = no data stored
broadcast datetime = no data stored
broadcast error = no data stored
broadcast hwcStatus = no data stored
broadcast id = no data stored
broadcast id = no data stored
broadcast id = no data stored
broadcast load = no data stored
broadcast outsidetemp = 18.000
broadcast queryexistence = no data stored
broadcast signoflife = no data stored
broadcast vdatetime = 11:39:00;28.08.2022
broadcast vdatetime = no data stored
general valuerange = no data stored
memory eeprom = no data stored
memory eeprom = no data stored
memory ram = no data stored
memory ram = no data stored
scan id = no data stored
scan.08  = Vaillant;BAI00;0701;3302
scan.08 id =  ;21;13;3700100088;4816;100053;27
scan.15  = Vaillant;SBX00;0326;4802
scan.15 id = 21;17;16;0020128487;0082;005147;N2

ebusctl grab result
1008b5070108 / 026400 = 3
1008b5110100 / 089e010a001f000080 = 21
10feb505025c00 = 7
10feb508020900 = 2
1008b51302050a / 0101 = 1

**Comment by bugzke** [comment]
Hi @DiaMontPhil 
Did you ever find the right config? I have a similar setup (Bulex ThemaCondens + Examaster) and facing the same issues.

**Reply by DiaMontPhil** [reply]
Hi Bugzke,
I have not been able to spend time looking for a solution, but I have been thinking about it.
Maybe some day, I'll continue searching and will post my findings.

**Comment by svendar** [comment]
Hello
I didn't start yet but I have a Bulex Isotwin condens 25-30/B and a Bulex Mipro R Thermostat.

something i found in one of the pull request over here
https://github.com/john30/ebusd-configuration/pull/436

Bulex is called Saunier Duval in other countries.
Saunier Duval MiPro Sense SRC 720 regulator looks like copy of Vaillant SensoComfort VRC 720.
Maybe the Bulex Mipro R can be a copy of the Vaillant VRC 700?
